src/extension/intellisenseHelper.ts

```diff
@@ -126,4 +188,83 @@ export class IntellisenseHelper {
 
         return Q(isRestartRequired);
     }
+
+    /**
+     * Simple check to see if the TypeScript package is in the expected location (where we installed it)
+     */
+    private static isTypeScriptInstalled() : Q.Promise<boolean> {
+        let fileSystem: FileSystem = new FileSystem();
+        let installPath: string = path.join(IntellisenseHelper.getTypeScriptInstallPath(), "lib");
+        return Q.resolve(fileSystem.exists(installPath));
+    }
+
+    /**
+     * Checks for the existance of our installed TypeScript package, if it exists also determine its version
+     */
+    private static getInstalledTypeScriptVersion() : Q.Promise<InstallProps> {
+        return IntellisenseHelper.isTypeScriptInstalled()
+            .then((installed : boolean) => {
+                    let installProps: InstallProps = {
+                        installed: installed,
+                        version: ""
+                    };
+
+                if (installed === true) {
+                    Log.logInternalMessage(LogLevel.Debug, "TypeScript is installed - checking version");
+                    return IntellisenseHelper.readPackageJson()
+                    .then( (version: string) => {
+                        installProps.version = version;
+                        return Q.resolve(installProps);
+                    });
+                }
+                else
+                {
+                    return Q.resolve(installProps);
+                }
+            });
+    }
+
+    /**
+     * Read the package.json from the TypeScript install path and return the version if it's available
+     */
+    public static readPackageJson(): Q.Promise<string> {
+        let packageFilePath: string = path.join(IntellisenseHelper.getTypeScriptInstallPath(), "package.json");
+        let fileSystem = new FileSystem();
+
+        return fileSystem.exists(packageFilePath)
+            .then(function(exists: boolean): Q.Promise<string> {
+                if (!exists) {
+                    return Q.reject<string>("package.json not found at:" + packageFilePath);
+                }
+
+                return fileSystem.readFile(packageFilePath, "utf-8");
+            })
+            .then(function(jsonContents: string): Q.Promise<any> {
+                return JSON.parse(jsonContents);
```

Comment:
Since `JSON.parse` is synchronous (and actually returns `any` not `Q.Promise<any>`) you can just merge this into the next step in the promise chain as `const data = JSON.parse(jsonContents); return data.version;`